src/main/java/be/seeseemelk/mockbukkit/MockBukkit.java

```diff
@@ -267,6 +269,117 @@ public static void loadJar(@NotNull File jarFile) throws InvalidPluginException
+	 * to load.
+	 *
+	 * @param <T>        The plugin's main class to load.
+	 * @param plugin     The plugin to load for mocking.
+	 * @param configStream The {@code InputStream} of the {@code config.yml} file to load.
+	 * @param parameters Extra parameters to pass on to the plugin constructor.
+	 * @return An instance of the plugin's main class.
+	 * @apiNote The File name can't be the same as the one the plugin loads by default
+	 */
+	public static <T extends JavaPlugin> @NotNull T loadWithConfig(
+			@NotNull Class<T> plugin,
+			InputStream configStream,
+			Object... parameters
+	)
+	{
+		FileConfiguration config = new YamlConfiguration();
+		try
+		{
+			config.load(new InputStreamReader(configStream));
+		}
+		catch (IOException | InvalidConfigurationException e)
+		{
+			throw new RuntimeException("Couldn't read config input stream",e);
+		}
+		return loadWithConfig(plugin, config, parameters);
+	}
+
+
+	/**
+	 * Loads and enables the Plugin with a specified Config File. It receives the {@code config.yml} to use as File
+	 * to load.
+	 *
+	 * @param <T>        The plugin's main class to load.
+	 * @param plugin     The plugin to load for mocking.
+	 * @param configFile The File of the {@code config.yml} file to load.
+	 * @param parameters Extra parameters to pass on to the plugin constructor.
+	 * @return An instance of the plugin's main class.
+	 * @apiNote The File name can't be the same as the one the plugin loads by default
+	 */
+	public static <T extends JavaPlugin> @NotNull T loadWithConfig(
+			@NotNull Class<T> plugin,
+			File configFile,
+			Object... parameters
+	)
+	{
+		FileConfiguration config = YamlConfiguration.loadConfiguration(configFile);
+		return loadWithConfig(plugin, config, parameters);
+	}
+
+	/**
+	 * Loads and enables the Plugin with a specified Config. It receives the {@code FileConfiguration} to load.
+	 *
+	 * @param <T>               The plugin's main class to load.
+	 * @param plugin            The plugin to load for mocking.
+	 * @param fileConfiguration The {@code FileConfiguration} to load.
+	 * @param parameters        Extra parameters to pass on to the plugin constructor.
+	 * @return An instance of the plugin's main class.
+	 * @apiNote The File name can't be the same as the one the plugin loads by default
+	 */
+	public static <T extends JavaPlugin> @NotNull T loadWithConfig(
+			@NotNull Class<T> plugin,
+			FileConfiguration fileConfiguration,
+			Object... parameters
+	)
+	{
+		ensureMocking();
+		PluginManagerMock pluginManager = mock.getPluginManager();
+		JavaPlugin instance = pluginManager.loadPlugin(plugin, parameters);
+
+		try
+		{
+			if (!instance.
+					getClass()
+					.getSuperclass()
+					.getSuperclass()
+					.getName()
+					.equals(BUKKIT_PLUGIN_SUPERCLASS))
+			{
+				throw new IllegalStateException("Invalid plugin superclass");
+			}
+			Field configField = instance
+					.getClass()
+					.getSuperclass()
+					.getSuperclass()
+					.getDeclaredField("newConfig");
```

Comment:
```suggestion
			Field configField = JavaPlugin.class.getDeclaredField("newConfig");
```
Why not do this instead?

Reply:
If i recall correctly, you actually need the specific instance of the field on a specific instance of a class for this to work

Reply:
Isn't there going to be only one class instance of JavaPlugin?

I don't think the class loader will create multiple duplicate instances of that class.

Reply:
Has this been tested? I do believe Thorinwasher's version should work since `getSuperclass` and `JavaPlugin.class` both just return a `Class`, and you pass in the desired instance when performing read/write operations. If this is the case (which it should be), the above check and `BUKKIT_PLUGIN_SUPERCLASS` can both be removed.

Reply:
It seems to work, but I fear it might break if you try loading multiple Plugins
But I guess thats something we have to look at down the road